src/diffusers/models/embeddings.py

```diff
@@ -37,10 +37,12 @@ def get_timestep_embedding(
     assert len(timesteps.shape) == 1, "Timesteps should be a 1d-array"
 
     half_dim = embedding_dim // 2
-    exponent = -math.log(max_period) * torch.arange(start=0, end=half_dim, dtype=torch.float32)
+    exponent = -math.log(max_period) * torch.arange(
+        start=0, end=half_dim, dtype=torch.float32, device=timesteps.device
```

Comment:
Do we need this opereration to run in fp32 even when the pipeline runs in fp16?

@patrickvonplaten @patil-suraj @anton-l

Reply:
Happy to try it out - let's maybe do it in a follow-up PR? :-)